src/agent.ts

```diff
@@ -1,45 +1,44 @@
-import {
-  Finding,
-  HandleTransaction,
-  TransactionEvent,
-  FindingSeverity, 
-  FindingType,    
-} from "forta-agent";
+import { Finding, HandleTransaction, TransactionEvent, FindingSeverity, FindingType } from "forta-agent";
 import { FORTA_BOT_REGISTRY, BOT_DEPLOYER_ADDRESS, NEW_AGENT_FUNCTION_SIGNATURE } from "./constants";
+import { Interface } from "@ethersproject/abi";
 
-
-export default function provideHandleTransaction(
-  proxyAddress: string,
-  deployerAddress: string,
+export function provideHandleTransaction(
+  fortaRegistryAddress: string,
+  nethermindDeployerAddress: string,
   functionABI: string
 ): HandleTransaction {
-  return async (txEvent: TransactionEvent):Promise<Finding[]>  => {
+  return async function handleTransaction(txEvent: TransactionEvent) {
     const findings: Finding[] = [];
     //check from address with deployer address
-    if (txEvent.from != deployerAddress.toLowerCase()) {
+    if (txEvent.from != nethermindDeployerAddress.toLowerCase()) {
+      return findings;
+    }
+    //check function ABI
+    const proxyInterface = new Interface([NEW_AGENT_FUNCTION_SIGNATURE]);
+    if (functionABI != NEW_AGENT_FUNCTION_SIGNATURE) {
```

Comment:
I think the block of code from L18-L21 may be left over from some tests you were running. It doesn't seem necessary.